libraries/systemsetup.rb

```diff
@@ -0,0 +1,29 @@
+module MacOS
+  module SystemSetup
+    def running_in_a_vm?
+      Chef.node['hardware']['machine_model'].match?(/Parallels/) # TODO: cover more hypervisors
+    end
+
+    def power_button_model?
+      Chef.node['hardware']['machine_model'].match? Regexp.union %w(Macmini MacPro)
+    end
+
+    def screensaver_disabled?
+      screensaver_settings('read', '0') &&
+        screensaver_settings('read-type', 'integer')
+    end
+
+    private
+
+    def screensaver_settings(query_type, expected_value)
+      expression_pattern = Regexp.new(expected_value)
+      shell_out('defaults', '-currentHost', query_type, 'com.apple.screensaver', 'idleTime',
```

Comment:
This should be it's own method so we can write unit tests for `screensaver_settings` more easily.